backend/src/GraphQl/OpenIdConnect/Application/ApplicationMutations.cs

```diff
@@ -0,0 +1,223 @@
+                OpenIddictConstants.Requirements.Features.ProofKeyForCodeExchange
+            }
+        };
+
+        var permissions = JsonSerializer.Deserialize<List<string>>(input.Permissions);
+        permissions?.ForEach(permission => descriptor.Permissions.Add(permission));
+
+        return new CreateApplicationPayload(await applicationManager.CreateAsync(descriptor, cancellationToken).AsTask().ConfigureAwait(false));
+    }
+
+    [UseUserManager]
+    [Authorize(Policy = AuthConfiguration.WritePolicy)]
+    public async Task<UpdateApplicationPayload> UpdateApplicationAsync(
+        UpdateApplicationInput input,
+        OpenIddictApplicationManager<OpenIddictEntityFrameworkCoreApplication> applicationManager,
+        ClaimsPrincipal claimsPrincipal,
+        UserManager<User> userManager,
+        ApplicationDbContext context,
+        CancellationToken cancellationToken
+    )
+    {
+        if (input.Id == null)
+        {
+            return new UpdateApplicationPayload(
+                new UpdateApplicationError(UpdateApplicationErrorCode.UNKNOWN,
+                "Empty Id",
+                Array.Empty<string>()));
+        }
+        if (!await OpenIdConnectAuthorization.IsAuthorizedToManageApplications(
+                claimsPrincipal,
+                userManager,
+                context,
+                cancellationToken
+            ).ConfigureAwait(false))
+        {
+            return new UpdateApplicationPayload(
+                new UpdateApplicationError(
+                    UpdateApplicationErrorCode.UNAUTHORIZED,
+                    "You are not authorized to update the application.",
+                    Array.Empty<string>()
+                )
+            );
+        }
+
+        var application = await applicationManager.FindByIdAsync(input.Id, cancellationToken).AsTask().ConfigureAwait(false);
+
+        if (application is null)
+        {
+            return new UpdateApplicationPayload(
+                new UpdateApplicationError(
+                    UpdateApplicationErrorCode.UNKNOWN_APPLICATION,
+                    "Unknown application.",
+                    new[] { nameof(input), nameof(input.Id).FirstCharToLower() }
+                )
+            );
+        }
+
+        var descriptor = new OpenIddictApplicationDescriptor();
+
+        await applicationManager.PopulateAsync(descriptor, application, cancellationToken).ConfigureAwait(false);
+
+        descriptor.ClientId = input.ClientId;
+        descriptor.DisplayName = input.DisplayName;
+        descriptor.RedirectUris.Clear();
+        descriptor.PostLogoutRedirectUris.Clear();
+        descriptor.RedirectUris.Add(new Uri(input.RedirectUri, UriKind.Absolute));
+        descriptor.PostLogoutRedirectUris.Add(new Uri(input.PostLogoutRedirectUri, UriKind.Absolute));
+        descriptor.Permissions.RemoveWhere(permission => permission.Contains("api"));
+        var permissions = JsonSerializer.Deserialize<List<string>>(input.Permissions);
```

Comment:
I would personally prefer a normal `foreach` loop

```
foreach (var permission in (JsonSerializer.Deserialize<List<string>>(input.Permissions)) ?? Array.Empty<string>())
{
    descriptor.Permissions.Add(permission)
}
```

In functional-style or LINQ-style proramming as used below with the `ForEach` method call, I'm prone to miss that it's not about doing something with the `permissions` variable but instead about the side-effect that is happening by calling `descriptor.Permissions.Add`. I miss that, because in functional languages there are **no** side-effects by language design. So using functional style with side-effects seems error-prone to me.